### PR TITLE
Feat: FAT-607-666: Resilient BLE scanning and BLE pairing flow as components

### PR DESCRIPTION
**Pull request by alexandremgo**
### 📝 Description

Resilient BLE scanning: at an arbitrary interval of time, the scanning is cleaned and restarted. 
Why ? When a previously paired device was exchanging messages via BLE just before the BLE scan starts, some exchange could still have been happening, making the device not available for scanning by our current library

Also: prepare the future remove of the current BlePairingFlow screen params: `successNavigateToConfig`, which is too complicated and at the end not enough flexible. On this PR: 
- move all the BLE pairing flow inside components
- keep the existing `screens/BleDevicePairingFlow/index.tsx` and its param, but using the newly created components

Also ++: 
- handle locked device response `0x5515` 
- new `HwTransportError` for all the implementations of Transport

### ❓ Context

- **Impacted projects**: `mobile` + `live-common` + `react-native-hw-transport-ble` + `hw-transport` + `errors`
- **Linked resource(s)**: [FAT-607](https://ledgerhq.atlassian.net/browse/FAT-607) + [FAT-666](https://ledgerhq.atlassian.net/browse/FAT-666) 😈
### ✅ Checklist

- [ ] **Test coverage** : unit test on hook + QA needed
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

See slack

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/BleDevicePairingFlow/BleDevicePairing.tsx [hunk 1]
```diff
@@ -22,7 +22,17 @@ export type BleDevicePairingProps = {
...
 };
 
-export const BleDevicePairing = ({
+/**
+ * Runs a BLE pairing with the given device. Displays pairing, success or error steps.
+ *
+ * A closing cross is displayed to the user during the pairing, which either calls onPaired if
+ * the device is already paired, or onRetry otherwise.
+ *
+ * @param deviceToPair Device to pair
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/BleDevicePairingFlow/BleDevicePairing.tsx:31]: are you sure?

Comment (alexandremgo) [reply apps/ledger-live-mobile/src/components/BleDevicePairingFlow/BleDevicePairing.tsx:31]: 🙈 📝

Comment (alexandremgo) [reply apps/ledger-live-mobile/src/components/BleDevicePairingFlow/BleDevicePairing.tsx:31]: it does mean i have good variable names 👑

File: apps/ledger-live-mobile/src/components/SelectDevice2/index.tsx [hunk 2]
```diff
@@ -39,9 +39,12 @@ type Navigation = BaseComposite<
 
 type Props = {
   onSelect: (_: Device) => void;
+  // Needed to know if a device has been selected and with which we are communicating with.
+  // The scanning can only occur when we are not communicating with a device.
+  isDeviceSelected: boolean;
```
Comment (grsoares21) [inline apps/ledger-live-mobile/src/components/SelectDevice2/index.tsx:44]: I personally would change the name of this prop so the users of the component can properly understand the impact it has. maybe "stopBleScanning" too, just like on the hook. wdyt?

**Review by grsoares21 — COMMENTED** [review]
Just a small comment. Otherwise LGTM

**Review by grsoares21 — APPROVED** [review]
LGTM. But please wait for QA ok to merge.

**Pull request merged**